hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -605,9 +607,8 @@ public final Stream<FileSlice> getLatestFileSlices(String partitionStr) {
       readLock.lock();
       String partitionPath = formatPartitionKey(partitionStr);
       ensurePartitionLoadedCorrectly(partitionPath);
-      return fetchLatestFileSlices(partitionPath)
-          .filter(slice -> !isFileGroupReplaced(slice.getFileGroupId()))
-          .map(this::filterBaseFileAfterPendingCompaction)
```

Comment:
Let's make signature like below:

```
Stream<FileSlice> filterBaseFileAfterPendingCompaction(FileSlice, boolean)
```

Then we just change `map` for `flatMap` here and everything else stays the same